### PR TITLE
Fixed blob method, added test

### PR DESCRIPTION
**Pull request by g4s8**
#10 - fixed blob method, added integration test to read and write same blob

### CODE REVIEW

**Comment by amihaiemil** [comment]
@g4s8 Can you assign it to me? :D

**Comment by 0crat** [comment]
This pull request [#48](https://github.com/artipie/docker-adapter/issues/48) is assigned to @amihaiemil[/z](https://www.0crat.com/u/amihaiemil), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/688580c0-1297-4552-a94a-c15f61b40680); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

**Comment by 0crat** [comment]
Job [#48](https://github.com/artipie/docker-adapter/issues/48) is already in scope

**Comment by 0crat** [comment]
Manual assignment of issues is discouraged, see [§19](http://www.zerocracy.com/policy.html#19): -5 point(s) just awarded to @g4s8[/z](https://www.0crat.com/u/g4s8)

File: src/main/java/com/artipie/docker/asto/AstoBlobs.java [hunk 1]
```diff
@@ -79,7 +80,7 @@ public AstoBlobs(final Storage asto) {
 
     @Override
     public CompletableFuture<Flow.Publisher<Byte>> blob(final Digest digest) {
-        return this.asto.value(new BlobRef(digest));
+        return this.asto.value(new Key.From(RegistryRoot.V2, new BlobRef(digest).string(), "data"));
```
Comment (amihaiemil) [inline src/main/java/com/artipie/docker/asto/AstoBlobs.java]: @g4s8 Can you indent this line a little? Something like this:

```java
return this.asto.value(
    new Key.From(
        RegistryRoot.V2,
        new BlobRef(digest).string(),
        "data"
    )
);
```

File: src/test/java/com/artipie/docker/storage/AstoBlobsITCase.java [hunk 2]
```diff
@@ -48,7 +49,8 @@
     @Test
     void saveBlobDataAtCorrectPath(@TempDir final Path tmp) throws Exception {
         final ByteArray blob = new ByteArray(new byte[]{0x00, 0x01, 0x02, 0x03});
-        final Digest digest = new AstoBlobs(new FileStorage(tmp)).put(
+        final BlobStore blobs = new AstoBlobs(new FileStorage(tmp));
```
Comment (amihaiemil) [inline src/test/java/com/artipie/docker/storage/AstoBlobsITCase.java]: @g4s8 Can you inverse the declaration of ``blobs`` and ``blob``? The "parent" should be delcared first so it's clearer.

File: src/test/java/com/artipie/docker/storage/AstoBlobsITCase.java [hunk 3]
```diff
@@ -69,4 +71,18 @@ void saveBlobDataAtCorrectPath(@TempDir final Path tmp) throws Exception {
             Matchers.equalTo(blob.primitiveBytes())
         );
     }
+
+    @Test
+    void writeAndReadBlob(@TempDir final Path tmp) throws Exception {
+        final ByteArray blob = new ByteArray(new byte[]{0x05, 0x06, 0x07, 0x08});
+        final BlobStore blobs = new AstoBlobs(new FileStorage(tmp));
```
Comment (amihaiemil) [inline src/test/java/com/artipie/docker/storage/AstoBlobsITCase.java]: @g4s8 Here as well, inverse their declaration please.

**Comment by g4s8** [comment]
@amihaiemil done, fixed in https://github.com/artipie/docker-adapter/pull/48/commits/555557d3491868d8efe046c562a253a3179cf0ce

**Comment by 0crat** [comment]
Job [#48](https://github.com/artipie/docker-adapter/issues/48) is not in the agenda of @amihaiemil[/z](https://www.0crat.com/u/amihaiemil), can't retrieve data and time of add

**Comment by amihaiemil** [comment]
@g4s8 see comment from the bot. I'm not sure if I will be paid or not. Can you reassign me?

**Comment by g4s8** [comment]
@0crat status

**Comment by 0crat** [comment]
Job was finished in 21 hours, bonus for fast delivery is possible (see [§36](http://www.zerocracy.com/policy.html#36))

**Comment by 0crat** [comment]
> @0crat status ([here](https://github.com/artipie/docker-adapter/issues/48#issuecomment-580137543))

@g4s8 This is what I know about this job in [CT2E6TK9B](https://www.0crat.com/p/CT2E6TK9B), as in [§32](http://www.zerocracy.com/policy.html#32):

  * The job [#48](https://github.com/artipie/docker-adapter/issues/48) is [in scope](http://www.zerocracy.com/policy.html#14) for 22hr
  * The role is `REV`
  * The job is not assigned to anyone
  * The [budget](http://www.zerocracy.com/policy.html#4) is not set yet
  * These users are banned and won't be assigned:
    * @g4s8[/z](https://www.0crat.com/u/g4s8): This user reported the ticket
  * Job [footprint](https://www.0crat.com/footprint/CT2E6TK9B?q=%7Bjob%3A%27gh%3Aartipie%2Fdocker-adapter%2348%27%2Ctype%3A%7B%24not%3A%2F%5ENotify%2F%7D%7D) (restricted area)

**Comment by 0crat** [comment]
Order was finished: +20 point(s) just awarded to @amihaiemil[/z](https://www.0crat.com/u/amihaiemil)

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @g4s8[/z](https://www.0crat.com/u/g4s8)

**Pull request merged**